controllers/ratelimitpolicy_controller.go

```diff
@@ -82,39 +85,56 @@ func (r *RateLimitPolicyReconciler) Reconcile(eventCtx context.Context, req ctrl
 		logger.V(1).Info(string(jsonData))
 	}
 
-	if rlp.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(rlp, rateLimitPolicyFinalizer) {
-		if err := r.finalizeRLP(ctx, rlp); err != nil {
-			return ctrl.Result{}, err
-		}
+	markedForDeletion := rlp.GetDeletionTimestamp() != nil
 
-		logger.Info("removing finalizer")
-		controllerutil.RemoveFinalizer(rlp, rateLimitPolicyFinalizer)
-		if err := r.UpdateResource(ctx, rlp); client.IgnoreNotFound(err) != nil {
+	// fetch the target network object
+	targetObj, err := r.FetchValidTargetRef(ctx, rlp.GetTargetRef(), rlp.Namespace)
```

Comment:
Maybe we could be explicit here and name it `targetNetworkObj` or simply `networkObj` to avoid confusion further in the code with other _targets_ , wdyt?